### PR TITLE
Transactions: don't expose mutable access to inner PSBTs

### PR DESCRIPTION
**Pull request by danielabrozzoni**
Signed-off-by: Antoine Poinsot <darosior@protonmail.com>
Co-Authored-by: Sebastian Geisler <sebastian@blockstream.io>
Co-Authored-by: Daniela Brozzoni <daniela@revault.dev>

### CODE REVIEW

File: src/transactions/mod.rs [hunk 1]
```diff
@@ -60,6 +60,27 @@ pub const TX_VERSION: i32 = 2;
...
+        /// Get a mutable reference to the inner transaction, this is only used internally
+        fn psbt_mut(&mut self) -> &mut Psbt;
+
+        /// Get a reference to the inner transaction, this is only used internally
+        fn psbt(&self) -> &Psbt;
+
+        /// Move inner transaction out
+        fn into_psbt(self) -> Psbt;
+
+        fn from_psbt_serialized(raw_psbt: &[u8]) -> Result<Self, TransactionSerialisationError>;
```
Comment (danielabrozzoni) [inline src/transactions/mod.rs:80]: @darosior I think we can expose those three methods (at least the into_psbt and from_psbt_serialized), no?

Comment (darosior) [reply src/transactions/mod.rs:80]: Yes we should actually!

**Comment by darosior** [comment]
Looks really good. Thanks Daniela and @sgeisler :)

Did you do some testing on downstream projects?

**Comment by darosior** [comment]
Also, please remove [`Signed-off-by`](https://developercertificate.org/) from the commit. It's really your work and Sebastian's, not mine!

**Comment by danielabrozzoni** [comment]
> Did you do some testing on downstream projects?

Nope, will do

**Comment by darosior** [comment]
reACK fe1e31f0427dab17741c6637c2b89de6aad51702

**Comment by danielabrozzoni** [comment]
After a bit of testing:
- `cosignerd`
https://github.com/revault/cosignerd/blob/ed0c9a5402aa0c9d20130de4584f7838539a8832/src/processing.rs#L95-L101
this seems easy to fix with `add_signature`
https://github.com/revault/cosignerd/blob/ed0c9a5402aa0c9d20130de4584f7838539a8832/src/processing.rs#L111-L135
I'm not sure about this one... it seems done that way to add signatures only at the end, when all the validation is done, but I'm not sure why we should do so.
- `revault_net`: seems ok
- `coordinatord`: seems ok
- `revaultd`: L O L
https://github.com/revault/revaultd/blob/master/src/daemon/control.rs#L791-L793
maybe we need a "clear_sigs" method or something like that?
A lot of `psbt_mut` used in the tests to add/remove inputs... It should be easy fixable

TL;DR: all good, maybe we need a "clear_signatures" method

**Comment by darosior** [comment]
Thanks for investigating. I'll take care of that
